generator/dryrun.py

```diff
@@ -16,6 +19,43 @@
 )
 
 
+def urlopen_with_retry(request, max_retries=5, initial_delay=2.0, timeout=30):
+    """
+    Perform urlopen with exponential backoff retry logic.
+
+    Args:
+        request: The urllib Request object
+        max_retries: Maximum number of retry attempts (default: 5)
+        initial_delay: Initial delay in seconds before first retry (default: 2.0)
+        timeout: Timeout in seconds for each request (default: 30)
+
+    Returns:
+        The response from urlopen
+
+    Raises:
+        URLError: If all retry attempts fail
+    """
+    last_error = None
+    delay = initial_delay
+
+    for attempt in range(max_retries + 1):
+        try:
+            return urlopen(request, timeout=timeout)
+        except URLError as e:
+            last_error = e
+            if attempt < max_retries:
+                print(
+                    f"Network request failed (attempt {attempt + 1}/{max_retries + 1}): {e}"
+                )
+                print(f"Retrying in {delay} seconds...")
+                time.sleep(delay)
+                delay *= 2  # Exponential backoff
+            else:
+                print(f"All {max_retries + 1} attempts failed")
+
+    raise last_error
```

Comment:
I noticed network errors when hitting the dryrun function, so implementing a retry mechanism here